p2p/net/swarm/swarm.go

```diff
@@ -602,6 +595,38 @@ func isBetterConn(a, b *Conn) bool {
 	return true
 }
 
+func connPriority(c *Conn) int {
+	if c == nil {
+		return 0
+	}
+
+	var priority int
+	// LAN > WAN > PROXY
+	switch {
+	case c.conn.Transport().Proxy():
+		priority += 10
+	case manet.IsPrivateAddr(c.RemoteMultiaddr()):
+		priority += 20
+	case manet.IsPublicAddr(c.RemoteMultiaddr()):
+		priority += 15
+	}
+
+	// We prefer udp protocols
+	switch c.ConnState().Transport {
+	case "quic", "quic-v1", "webtransport", "webrtc":
+		priority += 5
+	case "tcp", "websocket":
+		priority += 3
+	}
+
+	//  If one is transient and not the other, prefer the non-transient connection.
+	if !c.Stat().Transient {
+		priority += 2
```

Comment:
I think that should be in `isBetterConn` and not here.
Transient connections are fundamentally different than non transient one as applications can't and don't want to use theses for transferring data.

In other words, there is no score difference that will make us want to use a transient conn over a non transient one.
Even if the transient conn is quic-v1 over localhost and the non transient one is tcp over wan.
Because consumers (`go-libp2p-kad-dht`, `boxo/bitswap`, ...) wont use the transient connection over localhost to send and receive data.

Reply:
You make a good point, transient connections should be excluded from the beginning. Thanks for the advice❤, I will make changes later.